sources/cli.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+func consumeArgument(flag: String, from arguments: inout ArraySlice<String>) -> String? {
+    if let argumentIndex = arguments.index(of: flag) {
+        let value = arguments.suffix(from: argumentIndex).dropFirst().joined(separator: " ")
+        if value.isEmpty {
+            exitWithUsage()
+        }
+
+        arguments = arguments.prefix(upTo: argumentIndex)
+        return value
+    }
```

Comment:
Would `guard let` here to reduce indentation

Reply:
done